### PR TITLE
Debugger elements

### PR DESCRIPTION
**Pull request by Anthony-Eid**
Refactor debugger_panel_items by moving variable list code to elements.rs and making VariableList it's own view

### CODE REVIEW

File: crates/dap/src/debugger_settings.rs [hunk 1]
```diff
@@ -6,13 +6,7 @@ use settings::{Settings, SettingsSources};
 #[derive(Serialize, Deserialize, JsonSchema, Clone, Copy)]
 #[serde(default)]
 pub struct DebuggerSettings {
-    /// Whether the breakpoints should be reused across Zed sessions.
```
Comment (RemcoSmitsDev) [inline crates/dap/src/debugger_settings.rs]: These comments should not be removed.

File: crates/project/src/project.rs [hunk 2]
```diff
@@ -1547,12 +1547,18 @@ impl Project {
...
-            cx.spawn(|_, _| async move { client.terminate().await })
-                .detach_and_log_err(cx)
+            cx.spawn(|project, mut cx| async move {
+                client.terminate().await.log_err();
+
+                project.update(&mut cx, |_, cx| {
+                    cx.emit(Event::DebugClientStopped(client.id()))
+                })
+            })
+            .detach_and_log_err(cx)
```
Comment (RemcoSmitsDev) [inline crates/project/src/project.rs]:
```suggestion
            cx.spawn(|_, _| async move { client.terminate().await })
                .detach_and_log_err(cx)
```

File: crates/project/src/project.rs [hunk 3]
```diff
@@ -1547,12 +1547,18 @@ impl Project {
         cx.emit(Event::DebugClientStopped(client_id));
 
         if !should_terminate {
-            return;
+            return cx.emit(Event::DebugClientStopped(client_id));
```
Comment (RemcoSmitsDev) [inline crates/project/src/project.rs]:
```suggestion
            return;
```

File: crates/debugger_ui/src/debugger_panel_item.rs [hunk 4]
```diff
@@ -773,20 +481,52 @@ impl DebugPanelItem {
...
-        match self.open_entries.binary_search(&entry_id) {
-            Ok(ix) => {
-                self.open_entries.remove(ix);
-            }
-            Err(ix) => {
-                self.open_entries.insert(ix, entry_id.clone());
-            }
-        };
+impl EventEmitter<Event> for DebugPanelItem {}
+impl EventEmitter<ItemEvent> for DebugPanelItem {}
```
Comment (RemcoSmitsDev) [inline crates/debugger_ui/src/debugger_panel_item.rs]:
```suggestion
```

**Review by RemcoSmitsDev — CHANGES_REQUESTED** [review]

File: crates/project/src/project.rs (hunk 2 above)
Comment (Anthony-Eid) [reply crates/project/src/project.rs]: fixed, thank you

File: crates/dap/src/debugger_settings.rs (hunk 1 above)
Comment (Anthony-Eid) [reply crates/dap/src/debugger_settings.rs]: Added back

File: crates/debugger_ui/src/debugger_panel_item.rs (hunk 4 above)
Comment (Anthony-Eid) [reply crates/debugger_ui/src/debugger_panel_item.rs]: fixed

**Pull request merged**